x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutor.java

```diff
@@ -49,10 +53,20 @@ private void tryLockingPolicy(String policyName) {
             throw new EsRejectedExecutionException("Policy execution failed. Policy execution for [" + policyName +
                 "] is already in progress.");
         }
+        if (policyExecutionPermits.tryAcquire() == false) {
```

Comment:
Should we put this at the begin of the method here? so we dont have to remove from the policy locks if this check fails?

Reply:
If we put this acquire method first, then if the policy locks fail we'll still have to release the max-executions permit we acquired, so I don't think there would be any structural change. The locking mechanism that I felt was most likely to fail was what I put first in the sequence of conditions, but I'm not particularly bound to the arrangement.

Reply:
Then its no diff... Id just put the one that would be more likely to happen first then heh